### PR TITLE
python3Packages.e3-core: v22.3.1 -> v22.4.0

### PR DESCRIPTION
**Pull request by heijligen**
## Description of changes
* Use one line per item in list
* Add autoPatchelfHook for included binaries
* Update to version 22.4.0

Not building as python3.12 package due to "python312Packages.pbr" marked as broken (https://github.com/NixOS/nixpkgs/pull/289893/commits/1d899c604a78f8a8a64bf840429a19d6d0fc592f)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by atalii** [comment]
Thanks for putting this together; I'll double-check in a little while. You're not trying to get Alire working well in nixpkgs by any chance, are you? I'm happy to help out if that is the case :)

**Comment by heijligen** [comment]
I want to get most AdaCore packages into nixos. (ada-language-server etc). alire is unfortunately not my priority.

e3-testsuite  https://github.com/NixOS/nixpkgs/pull/290245
gnatPackages scope https://github.com/NixOS/nixpkgs/pull/283123

**Review by atalii — APPROVED** [review]

**Review by atalii — CHANGES_REQUESTED** [review]
(sorry about the noise; I'm just trying to get GH to show I've requested changes!)

**Review by atalii — APPROVED** [review]

File: pkgs/development/python-modules/e3-core/default.nix [hunk 1]
```diff
@@ -24,26 +27,40 @@
...
 buildPythonPackage rec {
   pname = "e3-core";
-  version = "22.3.1";
+  version = "22.4.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "AdaCore";
-    repo = "e3-core";
+    repo = "${pname}";
```
Comment (mweinelt) [inline pkgs/development/python-modules/e3-core/default.nix:35]: We're not reusing these, as they are technically something semantically different.

Comment (heijligen) [reply pkgs/development/python-modules/e3-core/default.nix]: Done

**Pull request merged**